HIP/hip-376.md

```diff
@@ -0,0 +1,155 @@
+---
+hip: 376
+title: Support for Approve/Allowance and ERC20/ERC721 transferFrom in HTS precompile
+author: Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2022-03-02
+discussions-to: 
+updated: 
+requires: 206, 218
+---
+
+## Abstract
+
+Extend the functionality from HIP-218 for ERC20 and ERC721 funtions working on
+HTS tokens with support for approve/allowance and transferFrom functions.
+
+## Motivation
+
+The goal of this HIP is to enable users of the HSCS to interact with Hedera
+native tokens in a manner that is similar to how they would in other EVM based
+systems. 
+
+## Rationale
+
+The standard model of ERC-20 and ERC-721 interactions involves directly calling
+the account address. Solidity, however, does a code size check before calling a
+smart contract and reverts if it finds no code. To address this a simple proxy
+redirect is proposed to be added to each token contract accessed that will
+redirect the call (including information about what address was called). HSCS
+will report all token address have this precompile, permitting solidity to
+interact with the token addresses as though it was an account with code.
+
+## User stories
+
+As a hedera smart contract services user, I want to be able to access my HTS
+fungible tokens like an Ethereum ERC-20 token.
+
+As a hedera smart contract services user, I want to be able to access my HTS
+non-fungible tokens like an Ethereum ERC-721 token.
+
+As a hedera smart contract services user, I want to be able to access metadata
+provided by the getTokenInfo in a smart contract.
+
+## Specification
+
+### EVM Precompile
```

Comment:
I would collapse this and the next two section into a reference to HIP-218. We are not creating new redirect contracts but are adding functions to the existing system contract.